DEVELOPMENT_WORKFLOW.rst

```diff
@@ -0,0 +1,138 @@
+====================================
+ RPC OpenStack Development Workflow
+====================================
+
+Summary
+=======
+
+It is currently necessary, to change the way the RPC OpenStack team develops
+the project (i.e., adds features, fixes bugs). The old workflow is described
+in the :ref:`Old Workflow <old-workflow>` section.
+
+The new workflow is outlined under :ref:`Proposed Workflow
+<proposed-workflow>`
+
+.. _old-workflow:
+
+RPC OpenStack's Old Workflow
+============================
+
+Facts
+-----
+
+#. |rpcos| is configured with the following branches:
+
+   - ``master``
+
+   - ``kilo``
+
+   When we decide to support liberty, a third branch will appear - ``liberty``.
+
+#. Bug fixes move from ``master`` to the other branch(es) via cherry-picks.
+
+#. Bugs are hand-managed by developers. Backport pull requests are submitted
+   and merged and the original bug is often not updated in a timely manner.
+
+#. Developers cannot use GitHub's commit message syntax to auto-close bugs.
+
+#. ``master`` 's current focus is not well defined. (Is it the next minor
+   release of the most recent stable branch? Is it the next major release of
+   RPC OpenStack?)
+
+Workflow
+--------
+
+#. A bug is filed or a feature is proposed to |rpcos|
+
+#. After triage and prioritization, a developer creates a fix.
+
+#. The fix is pushed to a branch on that developer's fork.
+
+   This prevents another developer from being able to collaborate on that
+   change (unless they're given access to that repository).
+
+#. The developer submits a pull request against ``master``. It's reviewed,
+   iterated upon, and eventually merged.
+
+#. If that PR is elligible to be backported, then the associated bug is
+   labeled as such and PRs are submitted to the stable branches.
+
+#. Stable PRs are then reviewed and merged.
+
+#. A developer will (eventually) close the original issue.
+
+.. _proposed-workflow:
+
+RPC OpenStack's Proposed New Workflow
+=====================================
+
+Pre-Workflow
+------------
+
+#. Add documentation to ``master`` that is easily discoverable detailing the
```

Comment:
@d34dh0r53 I'm not sure if master is still focused on kilo or if we're moving focus to liberty. If you can answer that, I can create a document in a separate PR to do this (or as part of this PR).